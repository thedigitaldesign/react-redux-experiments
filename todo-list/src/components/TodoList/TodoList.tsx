import React, { Component } from 'react'

export class TodoList extends Component {
  render() {
    return (
      <>
        TODO: Write looping logic based off redux state of todo's entered
      </>
    )
  }
}
