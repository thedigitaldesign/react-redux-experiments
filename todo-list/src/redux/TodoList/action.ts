// Types
import * as type from '../types'

export const GetTodoList = () => {
  return {
    type: type.TodoListTypes.TODO_LIST,
    payload: [{}],
  }
}
