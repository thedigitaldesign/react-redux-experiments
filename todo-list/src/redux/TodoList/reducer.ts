// Models
import { TodoListModel } from '../../models'

// Types
import * as type from '../types'

export default (state: any, action: { type: string; payload: TodoListModel | null }) => {
  switch (action.type) {
    case type.TodoListTypes.TODO_LIST:
      return action.payload?.list
    default:
      return state || null
  }
}
