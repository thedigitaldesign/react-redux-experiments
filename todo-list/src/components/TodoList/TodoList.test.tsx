import React from 'react'

// Packages
import Enzyme, { shallow } from 'enzyme'

// Routes
import { TodoList } from './TodoList'

describe('Components > TodoList', () => {
  const todoList = shallow(<TodoList />)

  it('should render components TodoList', () => {
    expect(todoList).toMatchSnapshot()
  })
})
