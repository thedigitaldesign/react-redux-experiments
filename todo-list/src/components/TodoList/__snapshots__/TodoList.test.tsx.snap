// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > TodoList should render components TodoList 1`] = `ShallowWrapper {}`;
